packages/web/pages/join/guild/[guildname].tsx

```diff
@@ -22,12 +23,101 @@ const SetupGuild: React.FC = () => {
 
   const guildName = router.query.guildname as string;
 
-  const { data, isValidating } = useSWR(guildName, getGuild);
+  const { data, isValidating } = useSWR(guildName, getGuild, {
+    revalidateOnFocus: false,
+  });
 
   useEffect(() => {
     setGuild(data);
   }, [data, guildName]);
 
+  const onSubmit = useCallback(
+    async (editGuildFormInputs: EditGuildFormInputs) => {
+      if (!guild) return;
+
+      const {
+        type,
+        discordAdminRoles: adminRoles,
+        discordMembershipRoles: membershipRoles,
+        logoFile,
+        logoUrl,
+        ...otherInputs
+      } = editGuildFormInputs;
+
+      let newLogoUrl = logoUrl;
+
+      if (logoFile?.[0]) {
+        const formData = new FormData();
+        formData.append('logo', logoFile[0]);
+
+        const result = await fetch(`/api/storage`, {
+          method: 'POST',
+          body: formData,
+          credentials: 'include',
+        });
+
+        const response = await result.json();
+        const { error, logo } = response;
+
+        if (result.status >= 400 || error || !logo) {
+          const description =
+            result.status >= 400 || error
+              ? `web3.storage ${result.status} response: "${
+                  error ?? result.statusText
+                }"`
+              : `Uploaded logo but didn't get a response back.`;
+          toast({
+            title: 'Error Saving Logo',
+            description,
+            status: 'warning',
+            isClosable: true,
+            duration: 8000,
+          });
+          return;
+        }
+        newLogoUrl = optimizedImage('logoURL', `ipfs://${logo}`);
+      }
```

Comment:
This is identical to the code in `edit`. Can it be moved to a helper?

Reply:
fixed in [6426825](https://github.com/MetaFam/TheGame/pull/1367/commits/64268253048e84ba01f12253f02fb02efa03a6f4)